elyra/pipeline/airflow/component_parser_airflow.py

```diff
@@ -46,149 +39,316 @@ class AirflowComponentParser(ComponentParser):
                                f"Proceeding as if '{data_type_info.data_type}' was detected.")
-            elif 'xcom' in arg.lower() and data_type_info.data_type == 'boolean':
+            elif 'xcom' in arg_name.lower() and data_type_info.data_type == 'boolean':
+                # Override a default of False for xcom push
                 data_type_info.default_value = True
 
-            # Override control id since all input properties can take in an xcom
-            control_id = "OneOfControl"
-
-            # Set the default control type
-            default_control_type = data_type_info.control_id
-
             # Create Dict of control ids that this property can use
-            one_of_control_types = [(default_control_type, data_type_info.data_type,
-                                    ControllerMap[default_control_type].value),
-                                    ("NestedEnumControl", "inputpath",
-                                    ControllerMap["NestedEnumControl"].value)]
-
-            component_params = ComponentParameter(id=arg,
-                                                  name=arg,
-                                                  data_type=data_type_info.data_type,
-                                                  value=(value or data_type_info.default_value),
-                                                  description=description,
-                                                  default_control_type=default_control_type,
-                                                  control_id=control_id,
-                                                  one_of_control_types=one_of_control_types,
-                                                  allow_no_options=True,
-                                                  required=required)
-            properties.append(component_params)
+            default_control_type = data_type_info.control_id
+            one_of_control_types = [
+                (default_control_type, data_type_info.data_type, ControllerMap[default_control_type].value),
+                ("NestedEnumControl", "inputpath", ControllerMap["NestedEnumControl"].value)
+            ]
+
+            component_param = ComponentParameter(
+                id=arg_name,
+                name=arg_name,
+                data_type=data_type_info.data_type,
+                value=(arg_attributes.get('default_value') or data_type_info.default_value),
+                description=description,
+                default_control_type=default_control_type,
+                control_id=CONTROL_ID,
+                one_of_control_types=one_of_control_types,
+                allow_no_options=True,
+                required=arg_attributes.get('required')
+            )
+            properties.append(component_param)
 
         return properties
 
-    def get_class_def_as_string(self, content: Dict[str, List[str]]) -> str:
+    def _get_init_arguments(self, init_function: ast.FunctionDef) -> Dict[str, Dict]:
```

Comment:
This method (and perhaps others) is geared completely to `python` (IIUC) - really cool.  Although not necessary now, we may want to consider introducing a base class to this one like `PythonParser` where these kinds of functions can live.  This can definitely wait until there's another platform that uses python component definitions, but something to think about in the background. 

I noticed "pieces" like `xcom` that are specific to Airflow, but perhaps those kinds of things could be abstracted into an override like `specific_keywords()` that the leaf implementations provide - assuming we can generalize those uses in a meaningful way (if that makes sense?).  This would then enable some of the other methods to be moved to a python-specific base class, for example.

Reply:
That is a good point and good to start thinking about it now. Especially with BYO runtimes and the open issue we have to support Kubeflow python components (in addition to just YAML-based)